### DISCUSSION TITLE
Problem building libmesh with intel compilers

### DISCUSSION THREAD

**Discussion by ke7kto** [idaholab/moose#19237] [Q&A General] [answered]
I recently installed the intel compilers on my Ubuntu 20.04 LTS machine. When recompiling MOOSE, I got errors. 

Here is my process:

    cd ~/my/new/moosedir/scripts
    ./update_and_rebuild_petsc.sh
    ./update_and_rebuild_libmesh.sh

This results in 

    /bin/ld: ./.libs/libmesh_opt.so: undefined reference to 
    vtkMPICommunicatorOpaqueComm::vtkMPICommunicatorOpaqueComm(int*)'

I've found `/usr/lib/x86_64-linux-gnu/libvtkParallelMPI-7.1.so` and the `libvtkParallelMPI-pv5.7.so` (ostensibly paraview vtk library) to have the definition of `_ZN28vtkMPICommunicatorOpaqueComm9GetHandleEv`, but am unsure how to proceed.

**Comment by roystgnr** [comment, marked as answer]
IIRC the Moose folks got sick of seeing issues from Intel compilers and have stopped supporting them, so you might want to give up on the project altogether.  If you don't, though, maybe open this as a libMesh issue?  I think we still have enough non-MOOSE icpc users to want to look into this.

You've got an undefined reference (is that the only one?) to a particular constructor there, but you're showing a definition for the GetHandle method on the same object, not the same thing.

I'm running 20.04 too, so I could probably reproduce this easily enough, though I don't currently have VTK installed.  Any particular reason you have VTK 7 installed rather than 6 or 9?

**Reply by roystgnr** [reply]
Oh, shoot, I don't actually have icpc installed on any of my current Ubuntu systems.  Sorry, I'm so used to having a system with intel modules handy that I'd forgotten that's no longer the case.  I might still be able to help you find a fix, but it'd be a lot slower if I can't replicate the problem.

**Reply by ke7kto** [reply]
Intel has released their compiler toolkits for free for general use under a "no support" tier. [Here's the instructions for adding the apt repo](https://www.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top/installation/install-using-package-managers/apt.html), but I made a mistake in adding the setvars.sh script to my `~/.bashrc`. I don't really need to use the intel tools to build moose.

There is no particular reason I have VTK 7 installed. I think it may have been autoselected for some reason.
